Fixed missing emails in the author section.

renderman21 opened the pull request:
As per #7936, here is the attempt in order to fix the missing emails as described. 

Some things to note, most of the email format have the characters `/dot/` and `/at` in them and that they had to be converted to the correct characters of `.` and `@` respectively. 

LADSPA plugins, however, are a different story. Originally, the solution was to manually change the email format but it did not work since it is dependent on the [ladspa](https://github.com/swh/ladspa/tree/0f54d2430febb4d5f02d13132dd91d7345e080b5) repository. Meaning, no changes can be made in , unless that repository is changed. Anyway, some author emails have a, rather, peculiar email format and that they've been manually changed. This case happens for the authors: Andy Wingo and Jesse Chappel. They both have spaces, and they are inconsistent with the other authors' emails present in the `plugins/swh` folder

renderman21 committed:
Convert author email to the correct format

renderman21 committed:
Merge branch 'LMMS:master' into author_email

renderman21 committed:
Fixed email formatting and able to show email

renderman21 committed:
Added conditional statements to replace specific author's emails with correct format

renderman21 committed:
Removed unnecessary lines

renderman21 committed:
Merge branch 'LMMS:master' into author_email

rubiefawn commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -78,10 +78,33 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
 	maker_content->setWordWrap( true );
 	l->addWidget( maker_label );
-	l->addWidget( maker_content, 1 );
+	l->addWidget( maker_content);
```
```suggestion
	l->addWidget(maker_content);
```

regulus79 commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -78,10 +78,33 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
 	auto maker_content = new QLabel(maker);
-	maker_content->setText( lm->getMaker( lkey ) );
+
+	auto makerString = lm->getMaker(lkey).toHtmlEscaped();
+
+	makerString.replace("&lt;", "<");
+	makerString.replace("&gt;", ">");
+	makerString.replace("&amp;", "&");
```
I'm curious why you first escape the text, but then seemingly un-escape it by converting the `&lt;`, `&gt;`, and `&amp;` to their normal symbols.

renderman21 replied:
I realized that this is redundant since the `setTextFormat()` is set to `Qt::PlainText`, meaning there's no need to do `toHtmlEscaped()`. It will be shown as is. Initially, I thought that the text format was set to `Qt::RichText` and that the entire thing would just not be shown, and needed to manually convert to their normal symbols. 

Will remove this if needed

regulus79 commented on src/gui/modals/EffectSelectDialog.cpp:
```diff
@@ -280,7 +280,12 @@ void EffectSelectDialog::rowChanged(const QModelIndex& idx, const QModelIndex&)
 			auto label = new QLabel(m_descriptionWidget);
 			QString labelText = "<p><b>" + tr("Name") + ":</b> " + QString::fromUtf8(descriptor.displayName) + "</p>";
 			labelText += "<p><b>" + tr("Description") + ":</b> " + qApp->translate("PluginBrowser", descriptor.description) + "</p>";
```
Should we also html escape the description of the plugin?

renderman21 commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -78,10 +78,33 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
+	makerString.replace("&gt;", ">");
+	makerString.replace("&amp;", "&");
+
+	if (makerString.contains("Andy Wingo"))
+	{
+		int startString = makerString.indexOf("<");
+		int endString = makerString.indexOf(">");
+		makerString.replace(startString + 1, endString - startString - 1, "wingo@pobox.com");
+	}
+	
+	if (makerString.contains("Jesse Chappel"))
+	{
+		int startString = makerString.indexOf("<");
+		int endString = makerString.indexOf(">");
+		makerString.replace(startString + 1, endString - startString - 1, "jesse@essej.net");
+	}
```
I admit that this is a _crude_ solution. Since I saw that most of the listed emails have the same format, I thought that these are special cases and it had to be manually written. 

Will optimize this by replacing `" at "` and ` " dot "`  to their normal symbols.

renderman21 committed:
Simplified solution on LadspaSubPluginFeatures

rubiefawn committed:
Delete missing space

messmerd reviewed (approved):
Looks good code-wise

rubiefawn committed:
Add QSizePolicy::Expanding to plugin description text

It wasn't utilizing the full width or height, which was especially apparent when resizing the window

rubiefawn committed:
Change LADSPA plugin description format

This makes it more stylistically consistent with the native plugins by using rich text, and also fixes the stupid text wrapping issue.

messmerd commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -62,68 +62,55 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
+				"host elects to use the same data location for both "
+				"input and output."
+			) + "</p>";
+	}
```
What do you think about moving these explanations to tooltips and leaving the label text as a simple yes/no?

rubiefawn replied:
I'm not sure. I have concerns about showing these properties to the user in general.

The only one of these that are ever shown to a user by plugins we bundle with LMMS is the "Real-time Capable" one, which feels like useless information, since AFAIK at this point in time LMMS itself is not real-time capable.

I think I have this backwards, and plugins that are *not* real-time capable should have a warning to say as much, and plugins that are real-time capable should not bother the end-user with things that should be the default.

Regarding "Real-time Dependency", how useful is this information to the user? The user certainly does not control whether the output of the plugin is cached. The user could potentially subject the plugin to significant latency I suppose, but I don't really know what that looks like in practice since we don't ship any LADSPA plugins with this property.

Regarding "In-place Broken", I genuinely don't think we should be showing the user this at all. It is completely out of their control. The end-user absolutely does not control buffer addresses. This is noise.

messmerd commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -62,68 +62,55 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
+				.replace(" at ", "@")
+				.replace(" dot ", ".")
+				.toHtmlEscaped()
+			+ "</p>"
```
```suggestion
		+ QWidget::tr("<p><b>Author:</b> %1</p>").arg(
			lm->getMaker(lkey).replace(" at ", "@").replace(" dot ", ".").toHtmlEscaped())
```

messmerd commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -62,68 +62,55 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
+				.replace(" dot ", ".")
+				.toHtmlEscaped()
+			+ "</p>"
+		"<p><b>" + QWidget::tr("Copyright") + ":</b> " + lm->getCopyright(lkey) + "</p>"
```
```suggestion
		+ QWidget::tr("<p><b>Copyright:</b> %1</p>").arg(lm->getCopyright(lkey))
```

rubiefawn commented on plugins/LadspaEffect/LadspaSubPluginFeatures.cpp:
```diff
@@ -62,68 +62,55 @@ void LadspaSubPluginFeatures::fillDescriptionWidget( QWidget * _parent,
+	const auto ldesc = lm->getDescription(lkey);
+
+	QString labelText =
+		"<p><b>" + QWidget::tr("Name") + ":</b> " + lm->getName(lkey) + "</p>"
```
I agree regarding the colon and spacing, but I don't think including the markup in the translation string makes sense. What about something like:

```suggestion
		"<p><b>" + QWidget::tr("Name: ") + "</b>" + lm->getName(lkey) + "</p>"
```

messmerd replied:
My concern was that an Arabic translation would look like this:
> الاسم: Allpass delay line

with the "Allpass delay line" text being bold while the name label is non-bold.

The only easy way around this issue I think would be to include the markup in the translation string.

rubiefawn committed:
Use `QString::arg()` instead of concatenation

I renounce the savagery of `string + string + string + string...`

rubiefawn committed:
Give EffectSelectDialog the same treatment

Also HTML-escape the plugin description as suggested by regulus79

rubiefawn commented on src/gui/modals/EffectSelectDialog.cpp:
```diff
@@ -277,14 +278,19 @@ void EffectSelectDialog::rowChanged(const QModelIndex& idx, const QModelIndex&)
+			}.arg(
+				tr("Name: "), descriptor.displayName,
+				tr("Description: "), qApp->translate("PluginBrowser", descriptor.description).toHtmlEscaped(),
+				tr("Author: "), QString::fromUtf8(descriptor.author)
```
Should the author come before the description?

**Edit:** I put the author before the description to be more consistent with the LADSPA dialog.

rubiefawn committed:
Remove GUI display for LADSPA in-place broken

rubiefawn committed:
Move markup inside translation strings

rubiefawn committed:
Put markup in translation strings (again)

rubiefawn committed:
Put author before description

to be more consistent with LADSPA

rubiefawn commented:
@Monospace-V has tested this, so I'll go ahead and merge 🤷‍♀️

<img width="919" height="449" alt="image" src="https://github.com/user-attachments/assets/a8858976-2ff8-415e-987a-e37288655738" />
